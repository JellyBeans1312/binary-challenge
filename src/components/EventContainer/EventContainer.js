import React from 'react';
import { connect } from 'react-redux'; 
import { Event } from '../Event/Event'

export const EventContainer = ({ events }) => {
  const eventCards = events.map(event => {
    return <Event 
      title={event.title}
      description={event.description}
      category={event.category}
      location={event.location}
      address={event.entities.formatted_address}
    />
  })
  return ( 
    eventCards
  )
}

export const mapStateToProps = state => ({
  events: state.events
})

export default connect(mapStateToProps)(EventContainer)