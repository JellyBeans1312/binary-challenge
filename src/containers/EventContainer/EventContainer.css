.event-components {
  height: 85%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  justify-items: center;
  margin: 10px 0;
  color: #fff;
}