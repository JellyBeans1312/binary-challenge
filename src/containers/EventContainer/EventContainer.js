import React from 'react';
import { connect } from 'react-redux'; 
import { Event } from '../../components/Event/Event'

export const EventContainer = ({ events }) => {
  if(!events) {
    return null
  }
  const eventCards = events.map(event => {
    return <Event 
      title={event.title}
      description={event.description}
      category={event.category}
      location={event.location}
    />
  })
  return ( 
    eventCards
  )
}

export const mapStateToProps = state => ({
  events: state.events
})

export default connect(mapStateToProps)(EventContainer)