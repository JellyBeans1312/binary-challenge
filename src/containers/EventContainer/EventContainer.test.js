import React from 'react';
import { shallow } from 'enzyme';
import { EventContainer, mapStateToProps } from './EventContainer';

describe('EventContainer', () => {
  let wrapper;

  const mockEvents = [
    { title: "something", category: 'concert', location: [{lat: 23, long: 234}] }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <EventContainer 
        events={mockEvents}
      />
    )
  });

  it('should matchsnapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapStateToProps', () => {
    it('should return an array of events', () => {
      const mockState = {
        events: [{ title: "mirrorland"}]
      };
      const expected = {
        events: [{ title: "mirrorland"}]
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  })
});