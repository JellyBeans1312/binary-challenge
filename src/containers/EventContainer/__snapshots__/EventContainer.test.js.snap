// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventContainer should matchsnapshot 1`] = `ShallowWrapper {}`;
